pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The default.mixed device is PennyLane's standard qubit simulator for mixed-state computations.
+
+It implements the necessary :class:`~pennylane.Device` methods as well as some built-in
+qubit :doc:`operations </introduction/operations>`, providing a simple mixed-state simulation of
+qubit-based quantum circuits.
+"""
+
+import numpy as np
+from pennylane import QubitDevice
+from pennylane.operation import DiagonalOperation, Channel
+
+
+class DefaultMixed(QubitDevice):
+    """Default qubit device for performing mixed-state computations in PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        shots (int): Number of times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to 1000 if not specified.
+            If ``analytic == True``, the number of shots is ignored
+            in the calculation of expectation values and variances, and only controls the number
+            of samples returned by ``sample``.
+        analytic (bool): indicates if the device should calculate expectations
+            and variances analytically.
+    """
+
+    name = "Default mixed-state qubit PennyLane plugin"
+    short_name = "default.mixed"
+    pennylane_requires = "0.12"
+    version = "0.12.0"
+    author = "Xanadu Inc."
+
+    # operations = {
+    #     "BasisState",
+    #     "QubitStateVector",
+    #     "QubitUnitary",
+    #     "DiagonalQubitUnitary",
+    #     "PauliX",
+    #     "PauliY",
+    #     "PauliZ",
+    #     "MultiRZ",
+    #     "Hadamard",
+    #     "S",
+    #     "T",
+    #     "CNOT",
+    #     "SWAP",
+    #     "CSWAP",
+    #     "Toffoli",
+    #     "CZ",
+    #     "PhaseShift",
+    #     "RX",
+    #     "RY",
+    #     "RZ",
+    #     "Rot",
+    #     "CRX",
+    #     "CRY",
+    #     "CRZ",
+    #     "CRot",
+    #     "AmplitudeDamping",
+    #     "GeneralizedAmplitudeDamping",
+    #     "PhaseDamping",
+    #     "DepolarizingChannel",
+    #     "QubitChannel",
+    # }
```

Comment:
```suggestion
    # }
    
    operations = {}
```
Looks like we need this to pass tests